aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterClearTest.java

```diff
@@ -0,0 +1,210 @@
+     */
+    @Test
+    public void clearDeletesAndRecreatesDatabase() throws DataStoreException, IOException, InterruptedException {
+        assertDbFileExists();
+        assertEquals(0, fileObserver.createFileEventCount);
+        assertEquals(0, fileObserver.deleteFileEventCount);
+        BlogOwner blogger1 = createBlogger("Dummy Blogger Sr.");
+        BlogOwner blogger2 = createBlogger("Dummy Blogger Jr.");
+        //Save a record and check if it's there
+        adapter.save(blogger1);
+        assertRecordIsInDb(blogger1.getName());
+        //Verify observer is still alive
+        assertFalse(subscriberDisposableRef.get().isDisposed());
+        assertObserverReceivedRecord(blogger1);
+        //The ensures files are created at least 1 second apart
+        Thread.sleep(1000);
+
+        adapter.clear();
+        //Make sure file was deleted and re-created
+        assertEquals(1, fileObserver.createFileEventCount);
+        assertEquals(1, fileObserver.deleteFileEventCount);
+        assertDbFileExists();
+        //Verify observer is still alive
+        assertFalse(subscriberDisposableRef.get().isDisposed());
+
+        //Make sure the new file is writable
+        adapter.save(blogger2);
+        //Check the new record is in the database
+        //and the old record is not.
+        assertRecordIsInDb(blogger2.getName());
+        assertRecordIsNotInDb(blogger1.getName());
+        assertObserverReceivedRecord(blogger2);
+        //Terminate the adapter
+        adapter.terminate();
+        //Verify observer was disposed.
+        assertTrue(subscriberDisposableRef.get().isDisposed());
+
+    }
+
+    private BlogOwner createBlogger(String name) throws DataStoreException {
+        return BlogOwner.builder()
+            .name(name)
+            .build();
+    }
+
+    private void assertObserverReceivedRecord(BlogOwner blogger) {
+        for (StorageItemChange<? extends Model> owner : observer.values()) {
+            if (owner.item() instanceof BlogOwner &&
+                blogger.getName().equals(((BlogOwner) owner.item()).getName())) {
+                return;
+            }
+        }
+        fail("Could not find " + blogger + " in event observer.");
+    }
+
+    private void assertRecordIsInDb(String name) throws DataStoreException {
+        List<BlogOwner> blogOwners = adapter.query(BlogOwner.class, BlogOwner.NAME.eq(name));
+        assertNotNull(blogOwners);
+        assertEquals(1, blogOwners.size());
+    }
+
+    private void assertRecordIsNotInDb(String name) throws DataStoreException {
+        List<BlogOwner> blogOwners = adapter.query(BlogOwner.class, BlogOwner.NAME.eq(name));
+        assertNotNull(blogOwners);
+        assertEquals(0, blogOwners.size());
+    }
+
+    private void assertDbFileExists() {
+        assertTrue(context.getDatabasePath(DATABASE_NAME).exists());
+    }
+
+    private static final class TestFileObserver extends FileObserver {
```

Comment:
Ah, cool beans. I've never seen or used the [`FileObserver`](https://developer.android.com/reference/android/os/FileObserver), before! Good stuff.

Reply:
Yeah...I initially used a method that would return me the created date of the file as an instance of `FileTime`, but it appears creation date was not available as an attribute in older versions of Android.